### DISCUSSION TITLE
Connection string dialog on app.planetscale.com showing is generating connection string for the wrong branch

### DISCUSSION THREAD

**Discussion by mbifulco** [planetscale/discussion#462] [Bug Reports] [answered]
Hi there,

In my planetscale org, I have several branches:

```
main
 |
 + feature-dev
    | 
    + working-branch-1
    + working-branch-2
```

`main` is my production branch, and `feature-dev` is a WIP branch that contains schema changes for a new feature we're rolling out. `feature-dev` also has several `working-branch-N` branches, where individual devs on my team are working on their updates to the schema for the feature as it's being built.

I'm trying to get a connection string to `working-branch-1`, so I go to app.planetscale.com, navigate to all branches, and click on `working-branch-1` - when I hit the `connect` button, it pops up a connection string dialog that is for a different branch -- in this case, the parent of the branch I'm looking at (`feature-dev`)... this has happened several times, and clearing my cache / logging out / in only seems to fix this sometimes. It's caused me to push changes to the wrong branch several times today. This is also happening for my teammates.

This appears to be a bug in the logic for that popup, if I had to guess.

Apart from what looks like a bug, I'd love to find a better way to do this - we're using DrizzleORM with a next.js/tRPC project to connect to planetscale, so I've been relying on this dialog to help me create an environment variable to connect to my DB from.

Is there a way to get these connection strings from the terminal with `pscale`?

<img width="1026" alt="Pasted_Image_5_8_23__2_24_PM" src="https://user-images.githubusercontent.com/1844496/236902217-872fccc7-c8ed-4876-b5c2-2312a35fd0c5.png">

**Comment by Michael-fore** [comment, marked as answer]
Create a new password and you can choose the branch  you want to create the account  for

**Reply by mbifulco** [reply]
Aha, so that actually helps quite a bit - each `password` is tied to a branch, and changing what is selected in the dropdown in this dialog indicates that as well. That's useful to know for sure, thanks @Michael-fore 🙌🏼 

In cases where the DB branch from the current _doesn't_ yet have a password associated with it, I'd expect this dialog to inform me of that - rather than show a connection string setup that might result in me pushing changes to the wrong branch.
